src/foam/nanos/ndiff/NDiffDAO.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+    package: 'foam.nanos.ndiff',
+    name: 'NDiffDAO',
+    extends: 'foam.dao.ProxyDAO',
+
+    doumentation: `
+    Decorator for an existing DAO that logs puts to the ndiffDAO service
+    if it's running. 
+    `,
+    javaImports: [
+        'foam.nanos.logger.Logger',
+        'foam.nanos.logger.Loggers',
+        'foam.nanos.ndiff.NDiff',
+        'foam.dao.DAO',
+        'foam.nanos.pm.PM',
+        'foam.core.X'
+    ],
+    javaCode: `
+        public NDiffDAO(foam.core.X x, foam.dao.DAO delegate) {
+            setX(x);
+            setDelegate(delegate);
+        }
+    `,
+    properties: [
+        {
+            name: 'nSpecName',
+            class: 'String'
+        },
+        {
+            name: 'runtimeOrigin',
+            class: 'Boolean'
+        }
+    ],
+    methods: [
+        {
+            name: 'put_',
+            javaCode: `
+            // object must be sent to the delegate dao first
+            // as there is a chance its ID has not been set
+            var storedObject = getDelegate().put_(x, obj);
+            
+            DAO ndiffDao = (DAO)x.get("ndiffDAO");
+            if ( ndiffDao == null ) {
+                return storedObject;
+            }
+            
+            PM pm = PM.create(x, this.getClass(), "put_");
+
+            Object objectId = storedObject.getProperty("id");
+            String nSpecName = getNSpecName();
+            
+            NDiff existingNdiff = (NDiff) ndiffDao.find(
```

Comment:
Since you've added
ids: [property1, property2]  to NDiff, you can now use the 
NDiffID FObject for your find. , look in your build/src/java for it.
Only if you think it helps.